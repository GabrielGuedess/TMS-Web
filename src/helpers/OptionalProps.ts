export type OptionalProps<T> = {
  [K in keyof T as `${string & K}`]?: T[K];
};
