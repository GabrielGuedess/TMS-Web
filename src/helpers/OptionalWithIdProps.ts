export type OptionalWithIdProps<T> = {
  [K in keyof T]: K extends 'id' ? T[K] : T[K] | undefined;
};
