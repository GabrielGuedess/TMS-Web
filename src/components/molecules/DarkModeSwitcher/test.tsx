import { ThemeProvider } from 'next-themes';

import userEvent from '@testing-library/user-event';
import { screen, waitFor } from '@testing-library/react';

import { render } from 'utils/tests/helpers';

import { DarkModeSwitcher } from '.';

describe('<DarkModeSwitcher />', () => {
  it('should render the dark mode switcher correctly', async () => {
    const { container } = render(
      <ThemeProvider>
        <DarkModeSwitcher />
      </ThemeProvider>,
      { theme: 'system' },
    );

    const button = screen.getByLabelText('switch');

    expect(button.getAttribute('aria-checked')).toBe('false');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the dark mode switcher light', async () => {
    const { container } = render(
      <ThemeProvider>
        <DarkModeSwitcher />
      </ThemeProvider>,
      { theme: 'light' },
    );

    const button = screen.getByLabelText('switch');

    expect(button.getAttribute('aria-checked')).toBe('false');

    await userEvent.click(button);

    await waitFor(() => {
      expect(button.getAttribute('aria-checked')).toBe('true');
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the dark mode switcher dark', async () => {
    render(
      <ThemeProvider>
        <DarkModeSwitcher />
      </ThemeProvider>,
      { theme: 'dark' },
    );

    const button = screen.getByLabelText('switch');

    expect(button.getAttribute('aria-checked')).toBe('true');

    await userEvent.click(button);

    await waitFor(() => {
      expect(button.getAttribute('aria-checked')).toBe('false');
    });
  });
});
