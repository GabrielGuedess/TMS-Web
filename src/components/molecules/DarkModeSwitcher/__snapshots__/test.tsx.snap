// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DarkModeSwitcher /> should render the dark mode switcher correctly 1`] = `
<script>
  !function(){var d=document.documentElement,c=d.classList;c.remove('light','dark');d.style.colorScheme = 'light';c.add('light')}()
</script>
`;

exports[`<DarkModeSwitcher /> should render the dark mode switcher light 1`] = `
<script>
  !function(){var d=document.documentElement,c=d.classList;c.remove('light','dark');d.style.colorScheme = 'light';c.add('light')}()
</script>
`;
